lib/workspace.rb

```diff
@@ -0,0 +1,68 @@
+require "httparty"
+require_relative "user"
+require_relative "channel"
+require "awesome_print"
+require "dotenv"
+
+Dotenv.load
+
+module Slack
+  class ResponseError < StandardError; end
+
+  class Workspace
+    BASE_URL = "https://slack.com/api/users.list"
+    attr_reader :users, :channels
+    attr_accessor :selection
+
+    def initialize
+      @users = User.list
+      @channels = Channel.list
+      @selection = nil
+    end
+
+    def select_user(name_or_id)
+      users.each do |user|
+        if name_or_id == user.name || name_or_id == user.slack_id || name_or_id == user.real_name
+          @selection = user
+        end
+      end
+      if @selection == nil
+        return "Sorry, #{name_or_id} is not a valid user."
+      end
+      return @selection
+    end
+
+    def select_channel(name_or_id)
```

Comment:
Do you notice how this and `select_user` are almost identical... that's a good sign you can dry it up with a helper method.